src/power_grid_model/core/serialization.py

```diff
@@ -80,7 +85,20 @@ def load(self) -> Dataset:
             A tuple containing the deserialized dataset in Power grid model input format and the type of the dataset.
         """
         pgc.deserializer_parse_to_buffer(self._deserializer)
-        return self._dataset.get_data()
+
+        filtered_data = self._dataset.get_data()
+        if self._data_filter is ComponentAttributeFilterOptions.RELEVANT:
+            for _, attributes in filtered_data.items():
```

Comment:
this only supports the all-component-level `RELEVANT` filter, but per-component filters should also be valid. The following `self._data_filter` should be valid:

```py
{
  ComponentType.node: None,
  ComponentType.line: ComponentAttributeFilterOptions.RELEVANT,
  ComponentType.sym_load: ComponentAttributeFilterOptions.ALL,
  ComponentType.asym_load: [],
  ComponentType.sym_gen: ["p_specified"],
  ComponentType.asym_gen: ["p_specified", "q_specified"],
}
```